### PR TITLE
[MM-50734] Add option to use German HPNS notification proxy

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
For paid customers, Mattermost offers two SLA-bound notification proxies. With this PR, the newly added proxy located in Germany can be selected via the System Console. Previously, customers had to enter the URL manually.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-50734

#### Screenshots
Screenshot from 2023-05-31 21-31-50
Screenshot from 2023-05-31 21-31-44

#### Release Note
```release-note
Add option to use German HPNS notification proxy
```

### CODE REVIEW

File: webapp/channels/src/components/admin_console/push_settings.jsx [hunk 1]
```diff
@@ -262,6 +289,30 @@ export default class PushSettings extends AdminSettings {
             );
         }
 
+        let locationDropdown;
+        if (this.state.pushNotificationServerType === PUSH_NOTIFICATIONS_MHPNS) {
+            const pushNotificationServerLocations = [];
```
Comment (pvev) [inline webapp/channels/src/components/admin_console/push_settings.jsx:294]: you can declare the array with the values directly, like `const arr = [{a: 'a'}, {b: 'b'}]`, no need to use the `Array.push` method.

File: webapp/channels/src/components/admin_console/push_settings.jsx [hunk 2]
```diff
@@ -53,6 +56,21 @@ export default class PushSettings extends AdminSettings {
             }
         }
 
+        if (id === 'pushNotificationServerLocation') {
+            switch (value) {
+            case PUSH_NOTIFICATIONS_LOCATION_US:
```
Comment (pvev) [inline webapp/channels/src/components/admin_console/push_settings.jsx:61]: nit: even when JS is optimized for switch case statements, for such simple case conditions like this one that doesn't even use a default, it is IMO for simplicity and readability to create a dictionary with the values and then assign the state directly:

```
const pushDic = {
    [PUSH_NOTIFICATIONS_LOCATION_US]: Constants.MHPNS_US,
    [PUSH_NOTIFICATIONS_LOCATION_DE]: Constants. MHPNS_DE,
 };
```
  ...
  
```
this.setState({
   pushNotificationServer: pushDic[value],
});
```

**Review by pvev — APPROVED** [review]
thank you @hanzei ! LGTM overall! some nits but nothing is really a blocker. 🚀

**Comment by pvev** [comment]
@hanzei one curious thing is that the spinwick sever is not taking your changes.

**Comment by pvev** [comment]
another thing I noticed, which can be handled as a different issue is that the formatting for the text below is not working:

Screenshot 2023-06-05 at 12 25 55

File: webapp/channels/src/components/admin_console/push_settings.jsx (hunk 2 above)
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/admin_console/push_settings.jsx:61]: i agree on @pvev but above switch state reads a lot easier to me

**Review by M-ZubairAhmed — APPROVED** [review]

File: webapp/channels/src/components/admin_console/push_settings.jsx (hunk 2 above)
Comment (hanzei) [reply webapp/channels/src/components/admin_console/push_settings.jsx:64]: @M-ZubairAhmed To which switch statement are you referring? The one newly added by me? Or the `if` `else` chain in https://github.com/mattermost/mattermost/blob/1de37de08093b4432f6b2f5de7ea427f0f48d323/webapp/channels/src/components/admin_console/push_settings.jsx#L45-L59?

**Comment by hanzei** [comment]
> another thing I noticed, which can be handled as a different issue is that the formatting for the text below is not working:

I was about to file a ticket for this separate issue but opted to fix it as part of this PR: [`d7cae09` (#23562)](https://github.com/mattermost/mattermost/pull/23562/commits/d7cae09fa62c2e2ec59a3664ff55ea81f5a52047)

File: webapp/channels/src/components/admin_console/push_settings.jsx (hunk 2 above)
Comment (hanzei) [reply webapp/channels/src/components/admin_console/push_settings.jsx:64]: @M-ZubairAhmed Could you please help me clarify your comment?

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/admin_console/push_settings.jsx:64]: Sorry missed this comment, i meant to say the switch statement which you added reads more clearly to be.

**Comment by hanzei** [comment]
@yasserfaraazkhan Would you be the right person to QA test this change?

**Comment by yasserfaraazkhan** [comment]
> @yasserfaraazkhan Would you be the right person to QA test this change?

@hanzei thank you, I can definitely try this on spinwick and update here

**Comment by yasserfaraazkhan** [comment]
Hi @hanzei 
thank you for the changes. The `Push Notification Server:` changes if user chooses `US`, `Germany`.

- There is one case where the `Push Notification Server:` value can be `push.mattermost.com`

**Steps**:

- Go to the System console.
- Go to Push Notification Server.
- Select "Use HPNS connection with...".
- Select `Germany` and confirm the value for `server` is changed
- from `Enable Push Notifications:` select some other option and switch back to `Use HPNS connection with uptime SLA`
- verify the `Server location:` is `Germany` and `Notification Server:` is `push.mattermost.com`


Screenshot 2023-06-08 at 1 48 00 AM


Screenshot 2023-06-08 at 1 42 04 AM


Screenshot 2023-06-08 at 1 43 19 AM

**Comment by hanzei** [comment]
Good catch @yasserfaraazkhan :+1: I've fixed the bug. Please take another look.

**Review by yasserfaraazkhan — APPROVED** [review]
thank you for the changes Ben.
was able to test this locally.

**Pull request merged**